### PR TITLE
🧪 Create basic E2E tests for Tarsier

### PR DESCRIPTION
**Pull request by asim-shrestha**

### CODE REVIEW

File: tarsier/tests/test_elements.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "html_file, expected_tag_to_xpath",
+    [
+        ("test_html/text_only.html", {0: "//html/body/h1", 1: "//html/body/p"}),
+        (
+            "test_html/hyperlink_only.html",
```
Comment (asim-shrestha) [inline tarsier/tests/test_elements.py:34]: Let's also have some expectation for page to text. (we can simplify the html to handle this and ensure there's only 1 /2 rows per test except the combination test

File: tarsier/tests/test_html/hyperlink_only.html [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Hyperlinks Only</title>
+</head>
+<body>
+
+    <p><a href="https://www.example.com" id="link1">Example Link 1</a></p>
```
Comment (asim-shrestha) [inline tarsier/tests/test_html/hyperlink_only.html:9]: Just one would suffice for the test

File: .gitignore [hunk 3]
```diff
@@ -153,3 +153,4 @@ cython_debug/
 *.d.ts
 *.js
 node_modules
+/tarsier/tests/test.env
```
Comment (asim-shrestha) [inline .gitignore:156]: */test.env

File: tests/test_elements.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+async def test_combined_elements_page(tarsier, html_file, expected_tag_to_xpath):
+    async with async_playwright() as p:
+        browser = await p.chromium.launch(headless=True)
+
+        html_file_path = os.path.abspath(
+            os.path.join(os.path.dirname(__file__), html_file)
+        )
+        context = await browser.new_context()
+        page = await context.new_page()
+        await page.goto(f"file://{html_file_path}")
```
Comment (awtkns) [inline tests/test_elements.py:96]: use the async_page() fixture here

File: tests/test_html/combination.html
Comment (awtkns) [inline tests/test_html/combination.html:1]: dont prefix with test here so pytest doesnt view it as an empty dir.
maybe `tests/mock_html/,,,` instead?

File: tests/test_elements.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+    [
+        ("test_html/text_only.html", {0: "//html/body/h1", 1: "//html/body/p"}),
+        (
+            "test_html/hyperlink_only.html",
+            {
+                0: '//html/body/p[1]/a[@id="link1"]',
+                1: '//html/body/p[2]/a[@id="link2"]',
+                2: '//html/body/p[3]/a[@id="link3"]',
+                3: '//html/body/p[4]/a[@id="link4"]',
+                4: '//html/body/p[5]/a[@id="link5"]',
```
Comment (asim-shrestha) [inline tests/test_elements.py:18]: Should just have 1 in these tests and combination should have like 3 things. Best to keep these as simple and small as possible

File: tests/test_elements.py [hunk 6]
```diff
@@ -0,0 +1,105 @@
+import os
+import pytest
+from playwright.async_api import async_playwright
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "html_file, expected_tag_to_xpath",
+    [
+        ("test_html/text_only.html", {0: "//html/body/h1", 1: "//html/body/p"}),
```
Comment (asim-shrestha) [inline tests/test_elements.py:10]: We should test against expected text

File: tests/conftest.py [hunk 7]
```diff
@@ -11,17 +11,32 @@
 from webdriver_manager.chrome import ChromeDriverManager
 
 from tarsier import GoogleVisionOCRService, Tarsier, MicrosoftAzureOCRService
+from dotenv import load_dotenv
 
+load_dotenv()
 IN_GITHUB_ACTIONS = os.getenv("GITHUB_ACTIONS") == "true"
```
Comment (asim-shrestha) [inline tests/conftest.py:17]: if no tin github actions, load dot env maybe. Or if node env is development

File: tests/test_elements.py [hunk 8]
```diff
@@ -0,0 +1,76 @@
+import os
+import pytest
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "html_file, expected_tag_to_xpath",
```
Comment (asim-shrestha) [inline tests/test_elements.py:7]: Let's also check expected text?

File: tests/test_elements.py [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+    [
+        ("mock_html/text_only.html", {0: "//html/body/h1", 1: "//html/body/p"}),
```
Comment (asim-shrestha) [inline tests/test_elements.py:9]: Could make it so thers only a h1 OR a p

File: tests/test_elements.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+    if "combination.html" not in html_file:
+        assert 2 < num_lines < 6, (
+            f"Number of lines in page_text does not meet the criteria for "
+            f"{html_file}. Got: {num_lines} lines."
+        )
+    else:
+        assert 9 < num_lines < 13, (
+            f"Number of lines in page_text does not meet the criteria for "
+            f"{html_file}. Got: {num_lines} lines."
+        )
```
Comment (asim-shrestha) [inline tests/test_elements.py:76]: Let's just assert that there exists some expected line on the page. This kind of testing is a bit sus (Not obvious why there should be between 2 and 6 lines in regular tests,  and why combination might be different)

**Review by awtkns — APPROVED** [review]

**Pull request merged**